packages/react-native-client/src/common/state.ts

```diff
@@ -0,0 +1,18 @@
+// hopefully temporary file with global state
+
+let isClientConnected = false;
+let isMicrophoneEnabled = false;
+
+export function isConnected() {
+  return isClientConnected;
+}
+export function setConnectionStatus(connected: boolean) {
+  isClientConnected = connected;
+}
+
+export function isMicrophoneSetToOn() {
+  return isMicrophoneEnabled;
+}
+export function setMicrophoneStatus(connected: boolean) {
+  isMicrophoneEnabled = connected;
+}
```

Comment:
Why do we need this? User already has `isMicrophoneOn` in `useMicrophone` and it comes from the native side.

Reply:
I wanted to make `useMicrophone` usable even before user starts a call. So there is no need to manage manual `setState` for microphone on preview screen. But native `isMicrophoneOn` can be used only if microphone is initialised (after call has started). So I made this hack for now, to make it possible. In future I would like to keep this state on native part - for now I avoided doing that as you and Karol are refactoring native code.

Reply:
You can use `useMicrophone` before the call (just like `useCamera` and `startCamera` btw) and we keep microphone state before the call. At least on Android and probably after Karol's refactor on iOS (that's why currently camera is kinda broken on iOS - the refactor enables camera to be started without a call and we're doing it now in example app). So I'd avoid doing temporary stuff like this just to delete it in two days.

Reply:
I've tried to do that, but if microphone is started in Android FishjamClient before joining room - warning is thrown that is it not possible to start microphone (AFAIR it because starting microphone is also starting new track, which can be done only if peer connection is established)

Reply:
There was a warning like this before my refactor, but now it shouldn't be, are you sure about this? 
Also you can start a new local track before establishing peer connection and we're currently doing that with camera.